internal/tiltfile/tiltfile_test.go

```diff
@@ -4381,25 +4381,317 @@ allow_k8s_contexts("hello")
+}
+
+func TestK8sResourceObjectInvalidSelector(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.setupFoo()
+	f.yaml("secret.yaml", secret("bar"))
+	f.yaml("namespace.yaml", namespace("baz"))
+
+	f.file("Tiltfile", `
+docker_build('gcr.io/foo', 'foo')
+k8s_yaml('foo.yaml')
+k8s_yaml('secret.yaml')
+k8s_yaml('namespace.yaml')
+k8s_resource('foo', objects=['baz:namespace:default:wot'])
+`)
+
+	f.loadErrString("Error making selector from string 'baz:namespace:default:wot'")
+}
+
+func TestK8sResourceObjectIncludesSelectorThatDoesntExist(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.setupFoo()
+	f.yaml("secret.yaml", secret("bar"))
+	f.yaml("namespace.yaml", namespace("baz"))
+
+	f.file("Tiltfile", `
+docker_build('gcr.io/foo', 'foo')
+k8s_yaml('foo.yaml')
+k8s_yaml('secret.yaml')
+k8s_yaml('namespace.yaml')
+k8s_resource('foo', objects=['baz:secret:default'])
+`)
+
+	f.loadErrString("Found 0 matches for baz:secret:default in remaining YAML. Object must match exactly 1 entity. Available YAML: bar:Secret:default, baz:Namespace:default")
+}
+
+func TestK8sResourceObjectsPartialNames(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.setupFoo()
+	f.yaml("secret.yaml", secret("bar"))
+	f.yaml("namespace.yaml", namespace("bar"))
+
+	f.file("Tiltfile", `
+docker_build('gcr.io/foo', 'foo')
+k8s_yaml('foo.yaml')
+k8s_yaml('secret.yaml')
+k8s_yaml('namespace.yaml')
+k8s_resource('foo', objects=['bar:secret', 'bar:namespace'])
+`)
+
+	f.load()
+	f.assertNextManifest("foo", deployment("foo"), k8sObject("bar", "Secret"), k8sObject("bar", "Namespace"))
+	f.assertNoMoreManifests()
+}
+
+func TestK8sResourcePrefixesShouldntMatch(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.setupFoo()
+	f.yaml("secret.yaml", secret("bar"))
+
+	f.file("Tiltfile", `
+docker_build('gcr.io/foo', 'foo')
+k8s_yaml('foo.yaml')
+k8s_yaml('secret.yaml')
+k8s_resource('foo', objects=['ba'])
+`)
+
+	f.loadErrString("Found 0 matches for ba in remaining YAML. Object must match exactly 1 entity. Available YAML: bar:Secret:default")
+}
+
+func TestK8sResourceAmbiguousSelector(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.setupFoo()
+	f.yaml("secret.yaml", secret("bar"))
+	f.yaml("namespace.yaml", namespace("bar"))
+
+	f.file("Tiltfile", `
+docker_build('gcr.io/foo', 'foo')
+k8s_yaml('foo.yaml')
+k8s_yaml('secret.yaml')
+k8s_yaml('namespace.yaml')
+k8s_resource('foo', objects=['bar'])
+`)
+
+	f.loadErrString("Found 2 matches for bar in remaining YAML. Object must match exactly 1 entity")
+}
+
+func TestK8sResourceObjectDuplicate(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.setupFoo()
+	f.yaml("secret.yaml", secret("bar"))
+	f.yaml("anotherworkload.yaml", deployment("baz"))
+
+	f.file("Tiltfile", `
+docker_build('gcr.io/foo', 'foo')
+k8s_yaml('foo.yaml')
+k8s_yaml('anotherworkload.yaml')
 k8s_resource('foo', objects=['bar'])
+k8s_resource('baz', objects=['bar'])
```

Comment:
hm, the fact that this test doesn't trip the "already claimed" error makes me think that the "already claimed" error happens specifically when _the same k8s_resource call_ references the same k8s object via two different selectors, is that true?

Reply:
Because we remove YAML from unresourced once we finish process one resource call, but not as we process one resource call, yes that's true.